packages/realm/bindgen/src/templates/typescript.ts

```diff
@@ -162,31 +162,21 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
 
   const spec = doJsPasses(bindModel(rawSpec));
 
-  const coreOut = file("core.ts", "eslint", "typescript-checker");
-  coreOut("// This file is generated: Update the spec instead of editing this file directly");
+  const out = file("native.d.mts", "eslint", "typescript-checker");
+  out("// This file is generated: Update the spec instead of editing this file directly");
 
-  coreOut("// Enums");
-  for (const e of spec.enums) {
-    // Using const enum to avoid having to emit JS backing these
-    coreOut(`export const enum ${e.jsName} {`);
-    coreOut(...e.enumerators.map(({ jsName, value }) => `${jsName} = ${value},\n`));
-    coreOut("};");
-  }
-  coreOut(`
+  out("declare module 'realm/binding' {");
```

Comment:
This is one of the more important changes.

Reply:
I think you are missing a word after "without"

Reply:
Thanks - I updated the comment 👍

Reply:
I still don't love that this file is picked up by the `src/*.ts` glob and injects its contents into a virtual module rather than being found via an `import` path. But I don't currently have a great idea for an alternative if we want to move away from rollup's replace plugin.